edn/src/types.rs

```diff
@@ -31,21 +43,33 @@ pub enum Value {
     NamespacedSymbol(symbols::NamespacedSymbol),
     Keyword(symbols::Keyword),
     NamespacedKeyword(symbols::NamespacedKeyword),
-    Vector(Vec<Value>),
+    Vector(Vec<ValueAndSpan>),
     // We're using a LinkedList here instead of a Vec or VecDeque because the
     // LinkedList is faster for appending (which we do a lot of).
     // See https://github.com/mozilla/mentat/issues/231
-    List(LinkedList<Value>),
+    List(LinkedList<ValueAndSpan>),
```

Comment:
I think this should remain `Value`, for current consumers.

Then, we want a parallel hierarchy like you've defined here, which could have two types (`ValueAndSpan(inner: InnerValueAndSpan , span: Span)` like you have, and `InnerValueAndSpan` where the container values are `ValueAndSpan`, just like here.

The map between is clear.